tracer/src/Datadog.Tracer.Native/fault_tolerant_tracker.cpp

```diff
@@ -165,7 +155,7 @@ bool fault_tolerant::FaultTolerantTracker::ShouldHeal(ModuleID moduleId, mdMetho
 
     if (shouldHeal)
     {
-        RequestRevert(moduleId, methodId, rejit_handler);
+        RequestRejit(moduleId, methodId, rejit_handler);
```

Comment:
The problem with this change and the difference between calling `Revert` and then `Rejit` is that: by calling `RequestRejit` without calling `RequestRevert` prior causes the `GetRejitParameters` event to receive the _current_ version of the bytecode (post-instrumentations) instead of the original bytecode on disk.
It will not only fail the fault-tolerant instrumentation, but also debugger's instrumentation. We rely on the fact of calling Revert & Rejit to perform our instrumentation on the original non-instrumented version of bytecode.
I would recommend calling Revert and then Rejit, but not omitting the `RequestRevert`.

Reply:
Are you sure about this? Documentation states that calling Rejit effectively reverts the code. See [here](https://learn.microsoft.com/en-us/dotnet/framework/unmanaged-api/profiling/icorprofilerinfo4-requestrejit-method), in remarks.
Also, the revert tests we have in CalTarget work.
Anyways, if an explicit call to revert is needed (as it was in the debugger code) we can explicitly call revert in the same call. But I don't want to add this kind of things if they are really not needed.

Reply:
We'll keep the discussion in https://github.com/DataDog/dd-trace-dotnet/pull/5533#discussion_r1605980510 to avoid duplication.